src/display/canvas.js

```diff
@@ -1280,9 +1282,49 @@ class CanvasGraphics {
     // transparent canvas when we have blend modes.
     const width = this.ctx.canvas.width;
     const height = this.ctx.canvas.height;
-
+    this.defaultBackgroundColor = background || "#ffffff";
     this.ctx.save();
-    this.ctx.fillStyle = background || "rgb(255, 255, 255)";
+
+    if (this.foregroundColor && this.backgroundColor) {
+      // Get the #RRGGBB value of the color. If it's a name (e.g. CanvasText)
+      // then it'll be converted to it's rgb value.
+      this.ctx.fillStyle = this.foregroundColor;
+      const fg = (this.foregroundColor = this.ctx.fillStyle);
+      this.ctx.fillStyle = this.backgroundColor;
+      const bg = (this.backgroundColor = this.ctx.fillStyle);
+
+      if (fg === "#000000" && bg === "#ffffff") {
+        this.foregroundColor = this.backgroundColor = null;
+      } else {
+        // https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_Colors_and_Luminance
+        //
+        // Relative luminance:
+        // https://www.w3.org/TR/WCAG20/#relativeluminancedef
+        //
+        // We compute the rounded luminance of the default background color.
+        // Then for every color in the pdf, if its rounded luminance is the
+        // same as the background one then it's replaced by the new
+        // background color else by the foreground one.
+        const cB = parseInt(this.defaultBackgroundColor.slice(1), 16);
```

Comment:
Wait, does this actually work correctly *in general* given what sort of values are listed as supported for the `background`-parameter?
Please refer to https://github.com/mozilla/pdf.js/blob/7d6d6f915104cc0d1d280d433e667ac722d30203/src/display/api.js#L1160-L1164

Reply:
I might be overlooking something here, but should we maybe even consider *deprecating* the old `background`-parameter in favour of the new `pageColors` one?

Reply:
If I'm correct, for Firefox, the `background` parameter is always null so finally the default bg is always white.
Consequently, we can add a check on the the default bg color using a `if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL"))`.
About this parameter in general, it has been added here:
https://github.com/mozilla/pdf.js/pull/8413
and in considering the use case shown in the 1st comment, I guess having the parameter is still relevant but in this case it's impossible to detect what is the real background color so the stuff I added is useless.
I'm wondering if I should rename the options I added here because it's really a patch about forced colors... something like `pageForcedColors`, wdyt ?

Reply:
Right, we probably need to keep the `background`-parameter.

---

I'm not sure that we actually need to rename new parameter.
However, we may want to *explicitly* skip the `background`-parameter (in the `canvas`-code) if `pageColors` have been set and also document that in the API?

Reply:
It isn't always skipped it depends.
For example if someone set it to a light gray then it'll be really a light gray for some people but for ones who enabled high contrast mode they will get the new colors.
And my patch is using the default bg luminance to "guess" when use forced bg or fg so the parameter never ignored (except when it isn't a #RRGGBB).

Reply:
OK, then how about simply adding a note to like this
```
 *
 *   NOTE: This option may be partially, or completely, ignored when the
 *   `pageColors`-option is used.
```
to the existing documentation at
https://github.com/mozilla/pdf.js/blob/7d6d6f915104cc0d1d280d433e667ac722d30203/src/display/api.js#L1160-L1164
and calling that good enough?